File: decompiled/3.20.2/be/ehealth/businessconnector/wsconsent/builders/RequestObjectBuilder.java
package be.ehealth.businessconnector.wsconsent.builders;

import be.ehealth.businessconnector.wsconsent.exception.WsConsentBusinessConnectorException;
import be.ehealth.technicalconnector.exception.TechnicalConnectorException;
import be.fgov.ehealth.hubservices.core.v2.AuthorWithPatientAndPersonType;
import be.fgov.ehealth.hubservices.core.v2.ConsentType;
import be.fgov.ehealth.hubservices.core.v2.GetPatientConsentRequest;
import be.fgov.ehealth.hubservices.core.v2.GetPatientConsentStatusRequest;
import be.fgov.ehealth.hubservices.core.v2.PutPatientConsentRequest;
import be.fgov.ehealth.hubservices.core.v2.RevokePatientConsentRequest;
import be.fgov.ehealth.hubservices.core.v2.SelectGetPatientConsentType;

public interface RequestObjectBuilder {
   PutPatientConsentRequest createPutRequest(AuthorWithPatientAndPersonType var1, ConsentType var2) throws TechnicalConnectorException, WsConsentBusinessConnectorException, InstantiationException;

   RevokePatientConsentRequest createRevokeRequest(AuthorWithPatientAndPersonType var1, ConsentType var2) throws TechnicalConnectorException, WsConsentBusinessConnectorException, InstantiationException;

   GetPatientConsentRequest createGetRequest(AuthorWithPatientAndPersonType var1, SelectGetPatientConsentType var2) throws TechnicalConnectorException, WsConsentBusinessConnectorException, InstantiationException;

   GetPatientConsentStatusRequest createGetStatusRequest(AuthorWithPatientAndPersonType var1, SelectGetPatientConsentType var2) throws TechnicalConnectorException, WsConsentBusinessConnectorException, InstantiationException;
}

File: src/main/kotlin/org/taktik/connector/business/wsconsent/builders/RequestObjectBuilder.kt
package org.taktik.connector.business.wsconsent.builders

import org.taktik.connector.business.wsconsent.exception.WsConsentBusinessConnectorException
import org.taktik.connector.technical.exception.TechnicalConnectorException
import be.fgov.ehealth.hubservices.core.v2.AuthorWithPatientAndPersonType
import be.fgov.ehealth.hubservices.core.v2.ConsentType
import be.fgov.ehealth.hubservices.core.v2.GetPatientConsentRequest
import be.fgov.ehealth.hubservices.core.v2.PutPatientConsentRequest
import be.fgov.ehealth.hubservices.core.v2.RevokePatientConsentRequest
import be.fgov.ehealth.hubservices.core.v2.SelectGetPatientConsentType

interface RequestObjectBuilder {
	@Throws(TechnicalConnectorException::class, WsConsentBusinessConnectorException::class, InstantiationException::class)
	fun createPutRequest(author: AuthorWithPatientAndPersonType?, consent: ConsentType?): PutPatientConsentRequest

	@Throws(TechnicalConnectorException::class, WsConsentBusinessConnectorException::class, InstantiationException::class)
	fun createGetRequest(author: AuthorWithPatientAndPersonType?, consent: SelectGetPatientConsentType?): GetPatientConsentRequest

	@Throws(TechnicalConnectorException::class, WsConsentBusinessConnectorException::class, InstantiationException::class)
	fun createRevokeRequest(author: AuthorWithPatientAndPersonType?, consent: ConsentType?): RevokePatientConsentRequest
}
